SpatialGDK/Source/SpatialGDK/Private/Tests/LoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,358 @@
+	const VirtualWorkerId IntentWorkerId,
+	const SpatialGDK::TSchemaOption<PhysicalWorkerName> ClientWorkerName = SpatialGDK::TSchemaOption<PhysicalWorkerName>())
+{
+	AddEntityToView(View, EntityId);
+
+	const WorkerRequirementSet ReadAcl;
+	WriteAclMap WriteAcl;
+	WriteAcl.Add(SpatialConstants::ENTITY_ACL_COMPONENT_ID, UnrealRequirementSet);
+	WriteAcl.Add(SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID, AuthRequirementSet);
+	WriteAcl.Add(SpatialConstants::COMPONENT_PRESENCE_COMPONENT_ID, AuthRequirementSet);
+	WriteAcl.Add(SpatialConstants::NET_OWNING_CLIENT_WORKER_COMPONENT_ID, AuthRequirementSet);
+
+	if (ClientWorkerName.IsSet())
+	{
+		const WorkerRequirementSet AuthClientRequirementSet = WorkerRequirementSet{ WorkerRequirementSet{
+			WorkerAttributeSet{ FString::Printf(TEXT("workerId:%s"), *ClientWorkerName.GetValue()) } } };
+		WriteAcl.Add(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID, AuthClientRequirementSet);
+		WriteAcl.Add(SpatialConstants::HEARTBEAT_COMPONENT_ID, AuthClientRequirementSet);
+	}
+
+	AddComponentToView(View, EntityId, MakeComponentDataFromData(SpatialGDK::EntityAcl(ReadAcl, WriteAcl).CreateEntityAclData()));
+	AddComponentToView(View, EntityId, MakeComponentDataFromData(SpatialGDK::AuthorityIntent::CreateAuthorityIntentData(IntentWorkerId)));
+	AddComponentToView(View, EntityId,
+					   MakeComponentDataFromData(SpatialGDK::ComponentPresence::CreateComponentPresenceData(PresentComponents)));
+	AddComponentToView(View, EntityId,
+					   MakeComponentDataFromData(SpatialGDK::NetOwningClientWorker::CreateNetOwningClientWorkerData(ClientWorkerName)));
+	AddComponentToView(View, EntityId, SpatialGDK::ComponentData(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID));
+	AddComponentToView(View, EntityId, SpatialGDK::ComponentData(SpatialConstants::HEARTBEAT_COMPONENT_ID));
+	AddComponentToView(View, EntityId, SpatialGDK::ComponentData(SpatialConstants::LB_TAG_COMPONENT_ID));
+
+	AddAuthorityToView(View, EntityId, SpatialConstants::ENTITY_ACL_COMPONENT_ID);
+}
+
+WriteAclMap GetWriteAclMapFromUpdate(const SpatialGDK::EntityComponentUpdate& Update)
+{
+	WriteAclMap AclMap;
+	Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(Update.Update.GetUnderlying());
+	// This is never emptied, so does not need an additional check for cleared fields
+	uint32 KVPairCount = Schema_GetObjectCount(ComponentObject, 2);
+	for (uint32 i = 0; i < KVPairCount; i++)
+	{
+		Schema_Object* KVPairObject = Schema_IndexObject(ComponentObject, 2, i);
+		uint32 Key = Schema_GetUint32(KVPairObject, SCHEMA_MAP_KEY_FIELD_ID);
+		WorkerRequirementSet Value = SpatialGDK::GetWorkerRequirementSetFromSchema(KVPairObject, SCHEMA_MAP_VALUE_FIELD_ID);
+
+		AclMap.Add(Key, Value);
+	}
+
+	return AclMap;
+}
+
+bool AclMapDelegatesComponents(const WriteAclMap& AclMap, const WorkerRequirementSet DelegatedRequirementSet,
+							   const TArray<Worker_ComponentId> DelegatedComponents)
```

Comment:
const ref to array might work nicely.

Reply:
done. Still working on my const ref muscle memory